src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -2,18 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.Syntax;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp.Syntax
 {
     internal static class SyntaxNodeRemover
     {
-        internal static TRoot RemoveNodes<TRoot>(TRoot root,
+        [return: NotNullIfNotNull("root")]
```

Comment:
Given that it's an error to substitute a nullable type for `TRoot` and `root` isn't nullable, can we just remove the annotation and attribute?

Reply:
Hm, it looks like I shouldn't have put this attribute. This method can return null if the `root` is one of the nodes to remove. Will correct